src/System.Numerics.Vectors/src/System/Numerics/Vector3.cs

```diff
@@ -270,17 +270,17 @@ public static Vector3 Clamp(Vector3 value1, Vector3 min, Vector3 max)
             // We must follow HLSL behavior in the case user specified min value is bigger than max value.
 
             float x = value1.X;
-            x = (x > max.X) ? max.X : x;
-            x = (x < min.X) ? min.X : x;
-
-            float y = value1.Y;
-            y = (y > max.Y) ? max.Y : y;
-            y = (y < min.Y) ? min.Y : y;
+            x = (min.X > x) ? min.X : x;  // max( x, minx)
```

Comment:
nit: It would be nice to fix the spacing on the `// max( x, minx)` comments here and in Vector4.cs as well